File: AdminApi/src/AdminApi.Infrastructure/DTOs/ClientOverview.cs
namespace Backbone.AdminApi.Infrastructure.DTOs;
public class ClientOverview
{
    public required string ClientId { get; set; }
    public required string DisplayName { get; set; }
    public required TierDTO DefaultTier { get; set; }
    public required DateTime CreatedAt { get; set; }
    public int? MaxIdentities { get; set; }
    public required int NumberOfIdentities { get; set; }
}

File: Applications/AdminUi/packages/admin_api_types/lib/src/clients/client_overview.dart
import 'package:json_annotation/json_annotation.dart';

import '../tiers/tier.dart';

part 'client_overview.g.dart';

@JsonSerializable()
class ClientOverview {
  final String clientId;
  final String displayName;
  final Tier defaultTier;
  final DateTime createdAt;
  final int? maxIdentities;
  final int numberOfIdentities;

  ClientOverview({
    required this.clientId,
    required this.displayName,
    required this.defaultTier,
    required this.createdAt,
    required this.maxIdentities,
    required this.numberOfIdentities,
  });

  factory ClientOverview.fromJson(dynamic json) => _$ClientOverviewFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$ClientOverviewToJson(this);
}
